open-telemetry/opentelemetry-js
# Update fetch instrumentation to be runtime agnostic

## Which problem is this PR solving?
This PR addresses https://github.com/open-telemetry/opentelemetry-js/issues/3413, but updating the experimental fetch instrumentation to not depend on browser specific constructs (window, location).

The motivation for this change is so I can instrument my NodeJS applications using `fetch()` with the `@opentelemetry/instrumentation-fetch` package.

My take is that this is much more of a bandaid than a true fix, as the fetch instrumentation still depends on the `opentelemetry-sdk-trace-web`, but it provides significant value in that I can use the instrumentation out of the box with no workarounds (like patching the global object `globalThis.location = {}`). I think the proper fix is to move some of the logic in `opentelemetry-sdk-trace-web` to `opentelemetry-sdk-trace-base`, but I am very new to this codebase and wanted to hold off on a larger refactor, in the interest of being able to contribute a targeted fix as quickly as possible.

Fixes # (issue)

## Short description of the changes

Check for the existence of browser-specific constructs before referencing them, if they are not present return undefined/don't evaluate the expression.

## Type of change

Please delete options that are not relevant.


## How Has This Been Tested?

I have run the tests and linting steps, and also tested this locally in a NodeJS application (fastify server that uses fetch).

My steps to test locally
* For both `experimental/packages/opentelemetry-instrumentation-fetch` and `packages/opentelemetry-sdk-trace-web`
* `cd` into that directory
* Build the package `npm run compile`
* Package it `npm pack`
* Install that tarball in my application `npm i ../../Desktop/opentelemetry-js/<path to tarball>.tgz`
* Run the application and jaeger locally 
* Cause it to make a fetch call
* See spans show up from the fetch instrumentation

## Checklist:
